llvm/include/llvm/IR/DebugInfoMetadata.h

```diff
@@ -2656,31 +2656,27 @@ class DIVariable : public DINode {
 /// DW_OP_stack_value) is the constant variable value.
 ///
 /// TODO: Co-allocate the expression elements.
-/// TODO: Separate from MDNode, or otherwise drop Distinct and Temporary
-/// storage types.
-class DIExpression : public MDNode {
+class DIExpression : public Metadata, ReplaceableMetadataImpl {
```

Comment:
I don't think DIExpression needs to inherit from ReplaceableMetadataImpl - it will never be temporary, and we never need to track its users or replace its uses.

Reply:
Should we document why DIExpression is not an MDNode here?